discovery/dns/dns.go

```diff
@@ -198,10 +213,20 @@ func (d *Discovery) refreshOne(ctx context.Context, name string, ch chan<- *targ
 			level.Warn(d.logger).Log("msg", "Invalid SRV record", "record", record)
 			continue
 		}
-		tg.Targets = append(tg.Targets, model.LabelSet{
-			model.AddressLabel: target,
-			dnsNameLabel:       model.LabelValue(name),
-		})
+		if validTxtRecord {
+			tg.Targets = append(tg.Targets, model.LabelSet{
+				model.AddressLabel:                target,
+				dnsNameLabel:                      model.LabelValue(name),
+				model.LabelName("__param_target"): targetLabel,
```

Comment:
This isn't something you should do in the code here. This makes the discovery method inflexible.

Instead, you should pass normal targets, and use relabel_configs to deal with the param.